### DISCUSSION TITLE
CLI and PerlMagick differ with -flip option

### DISCUSSION THREAD

**Discussion by bhelterline** [ImageMagick/ImageMagick#5984] [Help] [answered]
### ImageMagick version

7.1.0-57 Q16-HDRI x64

### Operating system, version and so on

Win10 x64

### Description

running the command
`c:\> magick file.png -flip file-flipped.png`

does not make the image upside down as described in the documents.  It appears to mirror it side to side like '-flop'

Running the same command in a perl script (the example.pl) does the correct thing.

**Comment by urban-warrior** [comment]
Both the CLI and PerlMagick make the same API call.  In addition, we created a white canvas with a small rectangle in each corner, each a different colors, so we could track the behavior.  The results for as expected, the flip turned the image upside down and the flop changed the left side with the right and visa-versa.

Post an example image here so we can try to reproduce your issue.   Before you do, convert your PNG to PNM.  Next flip the PNM.  Let's make sure the PNG does not have an orientation chunk that is skewing the results.

**Reply by bhelterline** [reply]
Results are the same with a .png and .pnm file.  I used the wizard logo
c:\temp\magick logo.png logo.pnm
c:\temp\magick logo.pnm -flip logo-flip.pnm

**Reply by bhelterline** [reply]
![logo](https://user-images.githubusercontent.com/2146021/213518555-f859a605-ec16-4af2-b5f3-d8daa091eab1.png)

It looks like .pnm files are not supported for upload :(

**Reply by dlemstra** [reply]
Adding `.txt` or zipping the file will allow you to upload it.

**Reply by bhelterline** [reply]
here are the .pmn files
[logo.pnm.txt](https://github.com/ImageMagick/ImageMagick/files/10460137/logo.pnm.txt)
[logo-flip.pnm.txt](https://github.com/ImageMagick/ImageMagick/files/10460139/logo-flip.pnm.txt)

**Reply by GeeMack** [reply]
I downloaded both images, "logo.pnm.txt" and "logo-flip.pnm.txt", deleted the ".txt" from the filenames, and opened them to view in Irfanview on Windows 11. The file "logo.pnm" is in the original upright orientation. The file "logo-flip.pnm" shows the same image vertically inverted, the expected behavior of a `-flip` operation. Reading them into ImageMagick commands also shows them as a properly oriented image and a flipped version of it.

**Reply by bhelterline** [reply]
You are correct.  The issues seems to be with the file.  My first attempt was from a photo on my phone and that did the wrong thing.  These images correctly flip.  It must have something to do with some metadata in the jpg file???

**Comment by fmw42** [comment, marked as answer]
Add -auto-orient to your command.  JPG files have EXIF information that includes how the camera was rotated when the picture was taken. It is ignored and will display the image wrong in general by viewers.  But if you re-orient the image before the flip with -auto-orient, it should end up correctly flipped relative to the re-oriented image.  You should re-orient and see what the correct orientation is when the image was captured so that you know what a proper flip will look like.
